server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteRoutingTableServiceIT.java

```diff
@@ -0,0 +1,260 @@
+        BlobPath indexRoutingPath = getIndexRoutingPath(baseMetadataPath.add(INDEX_ROUTING_PATH_TOKEN), indexRoutingTables.get(0).getIndex().getUUID());
+
+        AtomicInteger indexRoutingFiles = new AtomicInteger();
+        assertBusy(() -> {
+            indexRoutingFiles.set(repository.blobStore().blobContainer(indexRoutingPath).listBlobs().size());
+            assertTrue(indexRoutingFiles.get() > 0);
+        });
+
+        List<Long> routingTableVersions = getRoutingTableVersionsFromAllNodes();
+        assertTrue(areRoutingTableVersionsSame(routingTableVersions));
+
+        // Ensure node comes healthy after restart
+        String clusterManagerName = internalCluster().getClusterManagerName();
+        internalCluster().restartNode(clusterManagerName);
+        ensureGreen();
+        ensureGreen(INDEX_NAME);
+
+        // ensure master is elected and the cluster is stable
+        assertEquals(1, internalCluster().clusterService().state().nodes().getClusterManagerNode());
+        ensureStableCluster(4);
+
+        assertBusy(() -> {
+            int indexRoutingFilesAfterNodeDrop = repository.blobStore().blobContainer(indexRoutingPath).listBlobs().size();
+            assertTrue(indexRoutingFilesAfterNodeDrop > indexRoutingFiles.get());
+        });
+
+        RemoteManifestManager remoteManifestManager = remoteClusterStateService.getRemoteManifestManager();
+        verifyUpdatesInManifestFile(remoteManifestManager);
+    }
+
+    private BlobPath getBaseMetadataPath(BlobStoreRepository repository) {
+        return repository.basePath()
+            .add(Base64.getUrlEncoder().withoutPadding().encodeToString(getClusterState().getClusterName().value().getBytes(StandardCharsets.UTF_8)))
+            .add("cluster-state")
+            .add(getClusterState().metadata().clusterUUID());
+    }
+
+    private BlobPath getIndexRoutingPath(BlobPath indexRoutingPath, String indexUUID) {
+        RemoteStoreEnums.PathHashAlgorithm pathHashAlgo = RemoteStoreEnums.PathHashAlgorithm.FNV_1A_BASE64;
+        return pathType.path(
+            RemoteStorePathStrategy.PathInput.builder()
+                .basePath(indexRoutingPath)
+                .indexUUID(indexUUID)
+                .build(),
+            pathHashAlgo
+        );
+    }
+
+    private void verifyUpdatesInManifestFile(RemoteManifestManager remoteManifestManager) {
+        Optional<ClusterMetadataManifest> latestManifest = remoteManifestManager.getLatestClusterMetadataManifest(
+            getClusterState().getClusterName().value(),
+            getClusterState().getMetadata().clusterUUID()
+        );
+        assertTrue(latestManifest.isPresent());
+        ClusterMetadataManifest manifest = latestManifest.get();
+        assertTrue(manifest.getDiffManifest().getIndicesRoutingUpdated().contains(INDEX_NAME));
+        assertTrue(manifest.getDiffManifest().getIndicesDeleted().isEmpty());
```

Comment:
Lets also validate the indicesRouting paths added in manifest

Reply:
I see this comment is marked as resolved, but I didnt see verification being added. Can you point me to where this validation is added?

Reply:
Validation is added in below line, where we are checking `getIndicesRouting` method returns non empty response.

```
assertFalse(manifest.getIndicesRouting().isEmpty());
```

Reply:
Can we also verify the path details, ensuring we have path added for the index as expected.

Reply:
Added check to verify path in manifest.